### PR TITLE
refactor(serializers): convert serializers to ts

### PR DESCRIPTION
**Pull request by ahnpnl**
## Summary

- Convert all serializers to ts

- Move all serializers to a folder `serializers`

- Change `jest-preset.js` to reference to `serializers`

## Test plan

Green CI

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

One is using all `jest-preset-angular` snapshot serializers should declare:
```
const jestPresetAngularSerializers = require('jest-preset-angular/build/serializers')

module.exports = {
     // [...]
     snapshotSerializers: jestPresetAngularSerializers,
}
```

One is using one of `jest-preset-angular` snapshot serializers should declare:
```
module.exports = {
     // [...]
     snapshotSerializers: [
          'jest-preset-angular/build/serializers/html-comment'
     ]
}
```

## Other information
**N.A.**

### CODE REVIEW

File: src/serializers/ng-snapshot.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import type { Colors } from 'pretty-format';
+
+// jest hasn't exported this type yet so temporary we copy from 'pretty-format'
+interface PluginOptions {
+  edgeSpacing: string;
+  min: boolean;
+  spacing: string;
+}
+declare type Indent = (arg0: string) => string;
```
Comment (wtho) [inline src/serializers/ng-snapshot.ts:9]:
```suggestion
declare type Indent = (indentSpaces: string) => string;
```

File: src/serializers/ng-snapshot.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+declare type Print = (arg0: unknown) => string;
```
Comment (wtho) [inline src/serializers/ng-snapshot.ts:10]:
```suggestion
declare type Printer = (elementToSerialize: unknown) => string;
```

File: src/serializers/ng-snapshot.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  indent: Indent,
+  colors: Colors,
+  opts: PluginOptions,
+): string => {
+  return attributes
+    .sort()
+    .map((attribute) => {
+      return (
+        opts.spacing +
+        indent(colors.prop.open + attribute + colors.prop.close + '=') +
```
Comment (wtho) [inline src/serializers/ng-snapshot.ts:25]:
```suggestion
        indent(`${colors.prop.open}${attribute}${colors.prop.close}=`) +
```
I'm a fan of backslash-interpolation

File: src/serializers/ng-snapshot.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const { ɵivyEnabled } = require('@angular/core');
+
+  return !!ɵivyEnabled;
+};
+
+// Ivy component definition was stored on the `ngComponentDef`
+// property before `9.0.0-next.10`. Since `9.0.0-next.10` it was
+// renamed to `ɵcmp`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return
+const getComponentDef = (type: any): any => type.ngComponentDef || type.ɵcmp;
```
Comment (wtho) [inline src/serializers/ng-snapshot.ts:49]: Can we write basic return types for this?

File: src/serializers/ng-snapshot.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+const print = (fixture: any, print: Print, indent: Indent, opts: PluginOptions, colors: Colors): string => {
```
Comment (wtho) [inline src/serializers/ng-snapshot.ts:52]: I think we should give `fixture` some types
``` diff
- const print = (fixture: any, print: Print, indent: Indent, opts: PluginOptions, colors: Colors): string => {
+ const print = (fixture: ComponentFixture, print: Print, indent: Indent, opts: PluginOptions, colors: Colors): string => {
```

**Review by wtho — COMMENTED** [review]
I would like to eliminate most `any` (better now than later :sweat_smile:), but great work!

File: src/serializers/ng-snapshot.ts (hunk 1 above)
Comment (ahnpnl) [reply src/serializers/ng-snapshot.ts]: done

File: src/serializers/ng-snapshot.ts (hunk 2 above)
Comment (ahnpnl) [reply src/serializers/ng-snapshot.ts]: done

File: src/serializers/ng-snapshot.ts (hunk 3 above)
Comment (ahnpnl) [reply src/serializers/ng-snapshot.ts]: done

File: src/serializers/ng-snapshot.ts (hunk 5 above)
Comment (ahnpnl) [reply src/serializers/ng-snapshot.ts]: done

File: src/serializers/ng-snapshot.ts (hunk 4 above)
Comment (ahnpnl) [reply src/serializers/ng-snapshot.ts]: done

**Comment by ahnpnl** [comment]
I also remove most of any types for `no-ng-attributes.ts`

File: src/serializers/ng-snapshot.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+const test = (val: any): boolean =>
+  val !== undefined &&
+  val !== null &&
+  typeof val === 'object' &&
+  Object.prototype.hasOwnProperty.call(val, 'componentRef');
+
+export = {
```
Comment (ahnpnl) [inline src/serializers/ng-snapshot.ts:130]: we should convert to new plugin like `no-ng-attribute` at some point

File: src/serializers/html-comment.ts [hunk 7]
```diff
@@ -4,22 +4,15 @@
...
+const test = (value: any): boolean =>
+  value?.nodeType === 8 && value.constructor !== undefined && HTML_ELEMENT_REGEXP.test(value.constructor.name);
 
-const print = () => '';
+const print = (): string => '';
 
-module.exports = {
-  print: print,
-  test: test,
+export = {
```
Comment (ahnpnl) [inline src/serializers/html-comment.ts:15]: we should convert to new plugin like `no-ng-attribute` at some point

**Review by ahnpnl — COMMENTED** [review]
Would you please check again @wtho ?

**Review by wtho — APPROVED** [review]
Awesome! Thanks so much!

**Pull request merged**